### PR TITLE
[AVR-978] Add simulation inputs and outputs

### PR DESCRIPTION
**Pull request by lucabergamini**
Add simulation inputs and outputs to the `SimulationOutput` object

### CODE REVIEW

File: l5kit/l5kit/simulation/unroll.py [hunk 1]
```diff
@@ -117,6 +141,9 @@ def unroll(self, scene_indices: List[int]) -> List[SimulationOutput]:
         """
         sim_dataset = SimulationDataset.from_dataset_indices(self.dataset, scene_indices, self.sim_cfg)
 
+        agents_ins_outs: Dict[int, List[List[UnrollInputOutput]]] = {k: [] for k in scene_indices}
+        ego_ins_outs: Dict[int, List[UnrollInputOutput]] = {k: [] for k in scene_indices}
```
Comment (perone) [inline l5kit/l5kit/simulation/unroll.py:145]: I would use a `defaultdict` here.

File: l5kit/l5kit/simulation/unroll.py [hunk 2]
```diff
@@ -128,26 +155,44 @@ def unroll(self, scene_indices: List[int]) -> List[SimulationOutput]:
                     agents_input_dict = default_collate(list(agents_input.values()))
                     agents_input_dict = {k: v.to(self.device) for k, v in agents_input_dict.items()}
                     agents_output_dict = self.model_agents(agents_input_dict)
+
+                    # for update we need everything as numpy
+                    agents_input_dict = {k: v.cpu().numpy() for k, v in agents_input_dict.items()}
+                    agents_output_dict = {k: v.cpu().numpy() for k, v in agents_output_dict.items()}
```
Comment (perone) [inline l5kit/l5kit/simulation/unroll.py:161]: We do that very often in l5kit, I wonder if we shouldn't add a `move_to_numpy` or something similar, that can operate on dict values. Similar to the `move_to_gpu` we have.

File: l5kit/l5kit/simulation/unroll.py [hunk 3]
```diff
@@ -128,26 +155,44 @@ def unroll(self, scene_indices: List[int]) -> List[SimulationOutput]:
...
+
             # EGO
             if not self.sim_cfg.use_ego_gt:
                 ego_input = sim_dataset.rasterise_frame_batch(frame_index)
                 ego_input_dict = default_collate(ego_input)
                 ego_input_dict = {k: v.to(self.device) for k, v in ego_input_dict.items()}
                 ego_output_dict = self.model_ego(ego_input_dict)
+
+                ego_input_dict = {k: v.cpu().numpy() for k, v in ego_input_dict.items()}
+                ego_output_dict = {k: v.cpu().numpy() for k, v in ego_output_dict.items()}
```
Comment (perone) [inline l5kit/l5kit/simulation/unroll.py:179]: Same here.

**Review by perone — APPROVED** [review]
LGTM, just one comment regarding the numpy moves.

File: l5kit/l5kit/simulation/unroll.py (hunk 2 above)
Comment (lucabergamini) [reply l5kit/l5kit/simulation/unroll.py:161]: we don't have `move_to_gpu` either I think :(

Comment (lucabergamini) [reply l5kit/l5kit/simulation/unroll.py]: implemented both!

**Review by perone — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**